mod-source-record-manager-server/src/main/java/org/folio/verticle/periodic/AbstractPeriodicJobVerticle.java

```diff
@@ -0,0 +1,32 @@
+package org.folio.verticle.periodic;
+
+import io.vertx.core.AbstractVerticle;
+import org.folio.spring.SpringContextUtil;
+import org.springframework.context.support.AbstractApplicationContext;
+
+/**
+ * Abstract class that provides common info for running periodic jobs.
+ * Encapsulate all related spring stuff in this class.
+ * Subclasses will work only with business tasks.
+ */
+public abstract class AbstractPeriodicJobVerticle extends AbstractVerticle {
+  private static AbstractApplicationContext springGlobalContext;
+
+  //TODO: get rid of this workaround with global spring context
+  public static void setSpringGlobalContext(AbstractApplicationContext springGlobalContext) {
+    AbstractPeriodicJobVerticle.springGlobalContext = springGlobalContext;
+  }
+
+  @Override
+  public void start() {
+    context.put("springContext", springGlobalContext);
```

Comment:
I have seen this action a couple of times already. If `SprintContextUtil.autowireDependencies` wont work if there isn't an item called `springContext`, then i think this put should be wrapped around some other method. Success would mean that a user of `SpringContextUtil` would not need to know that `springContext` needs to be inserted into the vertx context.

Reply:
Refactored approach of initializing verticles. Now they initialized as spring beans. New SpringVerticleFactory has been introduced.

Reply:
The refactored approach is lovely.